syntax = "proto3";

package speech;

option go_package = "draw/pkg/speech/pb";

service SpeechService {
  rpc StreamTranscribe(stream TranscribeRequest) returns (TranscribeResponse);
  rpc CleanupSession(CleanupRequest) returns (CleanupResponse);
}

message TranscribeRequest {
  string session_id = 1;
  bytes audio_chunk = 2;
  bool end_of_stream = 3;
}

message TranscribeResponse {
  string transcription = 1;
  bool success = 2;
  string error = 3;
}

message CleanupRequest {
  string session_id = 1;
}

message CleanupResponse {
  bool success = 1;
}
